### ISSUE TITLE
Same-line inline annotations make phpcs remove comment marks, leading to invalid code

### ISSUE THREAD

**Issue by melbings** [squizlabs/PHP_CodeSniffer#2011]
I recently noticed this probable bug - phpcs analyzes the following snippet...

```
Debugger::getBar()->addPanel($this->objectFactory->make(DumperPanel::class, [
    'title' => 'SESS',
    'data' => \array_filter($_SESSION, /** @psalm-suppress MissingClosureParamType */ function (/** @noinspection PhpUnusedParameterInspection */ $value, string $key): bool {
        return $key !== 'user_password';
    }, ARRAY_FILTER_USE_BOTH)
]));
```

... and removes the leading `/**` of the second inline annotation, thus making the code invalid: 

```
Debugger::getBar()->addPanel($this->objectFactory->make(DumperPanel::class, [
    'title' => 'SESS',
    'data' => \array_filter($_SESSION, /** @psalm-suppress MissingClosureParamType */ function (@noinspection PhpUnusedParameterInspection */ $value, string $key): bool {
        return $key !== 'user_password';
    }, ARRAY_FILTER_USE_BOTH)
]));
```
Unfortunately, I have not had time to analyze this in any more detail, so that I'm providing the full code snippets instead of simplifying the code beforehand.

**Comment by gsherwood** [comment]
Just checking in to see if you can give me more info about your standard.

**Comment by melbings** [comment]
I'm sorry for keeping you waiting. I minimized the error condition down to

```
<?php

function(/** @demoAnnotation */ $value) {};
```

which outputs

```
<?php

function (@demoAnnotation */ $value) {
};
```

You can reproduce the problem by cloning https://github.com/melbings/CodeSnifferTest.

**Comment by gsherwood** [comment]
Thanks for doing that. I was able to replicate this bug and determine that it has already been fixed in master. The fix for this bug will be released in version 3.3.0.

While the bug report looks a little different, the fix was made as part of this report: https://github.com/squizlabs/PHP_CodeSniffer/issues/2006

Thanks again for taking the time to get that test case going - it made things really clear.
